### DISCUSSION TITLE
Feature request: Vertical tabs

### DISCUSSION THREAD

**Discussion by bstst** [zed-industries/zed#24006] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I would love to have the ability to switch the tab layout to vertical.

just like JetBrains do. this feature request is somewhat related to https://github.com/zed-industries/zed/issues/5240, yet it's not a request for a separate pane, only the ability to switch tab layout.

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://github.com/zed-industries/community/assets/412859/96ab233c-122b-4d1f-b81f-c2c5f5b9438d)

**Comment by syntastical** [comment]
Yeah, probably seems trivial to most, but vertical tabs has become a crucial part of my IDE needs.

**Comment by omid** [comment]
I'm just playing around with the source code to get more familiar with it.
I'm not sure if the zed devs will accept the changes or not, or how generally the process (I've read the CONTRIBUTING file) works? It's not a tiny change, it's also not a big one!

So far, it's like the screenshot below, but it's not resizable, and it's just _top_ or _right_ (and _none_, to remove one config? and I will try to add left and bottom also).
It's not a `pane`, it's exactly the same `tab_bar` we have on top.

Ref: https://github.com/zed-industries/zed/discussions/6667#discussioncomment-10025781 (and a bit https://github.com/zed-industries/zed/issues/5240)

![image](https://github.com/user-attachments/assets/e9ee2082-a85b-447e-951f-be63a710bfbc)

PS: the code is not close to ready to be a PR.

**Comment by m-czyz** [comment]
hey @omid could you somehow share your WIP changes? I would like to contribute to the development of this feature too, but I don't want to start from the scratch

**Comment by omid** [comment]
@m-czyz sure. But be aware that it's just a PoC.
For example, I completely commented out the top bar, instead of handling it by conditions.
I made a PR [here](https://github.com/omid/zed/pull/1) for [this](https://github.com/omid/zed/tree/feat/vertical_bar) branch.

PS: We can remove `tab_bar.show` config also. And add a `None` to `TabBarPosition` enum.

**Comment by HaroldAlingGSD** [comment]
Any progress on this? I actually was surprised this wasn't an option in Zed ...

**Comment by DmitrySkibitsky** [comment]
+1. It would be great to be able to display tabs vertically. Horizontal tab distribution is impractical. After opening several files, it becomes difficult to visually identify which file is open. There is “tab_switcher::Toggle,” but I would prefer to display tabs vertically in the editor window.

**Reply by DmitrySkibitsky** [reply]
I'm literally missing a few features in Zed, and I would say that this editor will replace JetBrains for me on Vue + TS projects.

**Reply by bstst** [reply]
Totally. I'm somewhat tired of JetBrains. Something lighter like Zed would cheer up the day. Yet I absolutely will not touch an editor that does not support vertical tabs.

Yet when I see this I feel that this feature won't be implemented soon:

<img width="369" height="55" alt="image" src="https://github.com/user-attachments/assets/c5fa2340-489d-4d52-a755-d2555bb54c58" />

**Comment by subhasish-smiles** [comment]
Hello I'm still hoping that this feature will be implemented.

**Comment by wintercounter** [comment]
The only thing preventing me moving to Zed.

**Comment by bstst** [comment]
Well. Created a PR for this (https://github.com/zed-industries/zed/pull/42717) — but not sure if this shall be ever merged (208 pending merge requests as of now).

But it's ok for me, rocking the dev build based on that PR on my machine and enjoying the zed with vertical tabs.

**Comment by hrgdavor** [comment]
+1 for this feature,

I am subscribed to this issue, to revisit zed editor once it is implemented.

**Comment by SomeoneToIgnore** [comment]
Given https://github.com/zed-industries/zed/pull/42717#issuecomment-3571456682 which mean nothing will happen before Zed's 1.0 at least, will keep all related things in https://github.com/zed-industries/zed/issues/5240 and lock this issue to avoid further nonsense +1 comments.
